src/Microsoft.ApiDesignGuidelines.Analyzers/Core/DoNotDecreaseInheritedMemberVisibility.cs

```diff
@@ -24,16 +27,81 @@ public abstract class DoNotDecreaseInheritedMemberVisibilityAnalyzer : Diagnosti
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182332.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         { 
-            
+            analysisContext.RegisterSymbolAction(CheckForDecreasedVisibility, SymbolKind.Method, SymbolKind.Field, SymbolKind.Property, SymbolKind.Event);
+        }
+
+        private void CheckForDecreasedVisibility(SymbolAnalysisContext context)
+        {
+            // Bail out if the member is publicly accessible, or sealed, or on a sealed type
+            if (IsVisibleOutsideAssembly(context.Symbol) || context.Symbol.IsSealed || (context.Symbol.ContainingType?.IsSealed ?? true))
```

Comment:
I actually haven't looked into the perf of nullableThing?.BooleanProperty ?? true/false. I just like the syntax because it seems a bit more compact while remaining readable. Would be interesting to see if the generated IL is different in any significant way or not.

Also, I've pulled context.Symbol out into a local.